blocks/datasets/streams.py

```diff
@@ -314,6 +314,28 @@ def _cache(self):
             cache.extend(data)
 
 
+class DataStreamSort(DataStreamMapping):
+    """Sorts the contents of the batches of the wrapped data stream.
+
+    Parameters
+    ----------
+    data_stream : instance of :class:`DataStream`
+        The wrapped data stream.
+    key : callable
+        The mapping that returns the value to sort on
+
+    """
+    def __init__(self, data_stream, key=None):
+
+        def mapping(x):
```

Comment:
Mm, nested functions aren't supported either by pickle (sorry, it's quite a nightmare sometimes).

What you could do is to create a callable `Mapping` class instead, which takes `key` and `reverse` as arguments to its `__init__` method. If you define this class in the global namespace, everything should be fine.